libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -248,7 +247,11 @@ class YulUtilFunctions
 	/// the specified slot and offset. If offset is not given, it is expected as
 	/// runtime parameter.
 	/// signature: (slot, [offset,] value)
-	std::string updateStorageValueFunction(Type const& _type, std::optional<unsigned> const& _offset = std::optional<unsigned>());
+	std::string updateStorageValueFunction(
+		Type const& _toType,
+		Type const* _fromType = nullptr,
+		std::optional<unsigned> const& _offset = std::optional<unsigned>()
+	);
```

Comment:
I'm not quite happy with this signature... if we want to keep it like this, we should document ``_fromType``.
IIUC ``_fromType`` currently ``_fromType```should be ``nullptr`` for all value types, but has to be a valid reference type, whenever ``_toType`` is a reference type...

Maybe the safest option would be to change the signature for this to ``updateStorageValueFunction(Type const& _fromType, Type const& _toType, ...)`` and assert ``_fromType == _toType`` if ``_toType.isValueType()``? And if ``_toType`` is a reference type maybe even also assert that ``toReferenceType.copyForLocationIfReference(&_fromType)`` is not null and the same as ``_toType``?

On the call sites of ``updateStorageValueFunction`` I guess we'll either have a type available to use as ``_fromType``, or we can just pass the same type as ``_toType`` to it...

Reply:
I updated signature to accept `Type const& _fromType, Type const& _toType, ...`. Also added asserts for both value and reference types.